core/src/main/java/org/apache/iceberg/deletes/DVFileWriter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+import java.io.Closeable;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.io.DeleteWriteResult;
+
+/** A deletion vector file writer. */
+public interface DVFileWriter extends Closeable {
+  /**
+   * Marks a position in a given data file as deleted.
+   *
+   * @param path the data file path
+   * @param pos the data file position
+   * @param spec the data file partition spec
+   * @param partition the data file partition
+   */
+  void write(CharSequence path, long pos, PartitionSpec spec, StructLike partition);
```

Comment:
Little bit of a nit here, but does `write` make sense?   We're actually marking for delete.  `FileAppender` and most "Writers" use `add`, should this just be `delete`?

Reply:
Agree. What about `DVFileWriter` as the name? That seems OK but if there are alternatives, let me know.

Reply:
@danielcweeks from my standpoint, `delete()` would be confusing. As the class is named `DeleteVectorFileWriter` (yes, I would prefer a full name instead of `DV`), I think it makes sense to use `write()` there.

Reply:
In my view, `delete` makes sense now as previously we passed `PositionDelete` object and used the common writer abstraction. That said, I am OK either way.

I do want to keep the shorter class name. I use `DV` naming consistently throughout the code and it allows me to stay on one line most of the time. Our 100 character line limit is no joke!